src/main/java/io/reactivex/Maybe.java

```diff
@@ -444,6 +857,25 @@
                 Functions.EMPTY_ACTION     // dispose
         ));
     }
+    
+    /**
+     * Calls the given onEvent callback with the (success value, null) for an onSuccess, (null, throwabe) for
+     * an onError or (null, null) for an onComplete signal from this Maybe before delivering said
+     * signal to the downstream.
+     * <p>
+     * Exceptions thrown from the callback will override the event so the downstream receives the
+     * error instead of the original signal.
+     * <dl>
+     * <dt><b>Scheduler:</b></dt>
+     * <dd>{@code doOnEvent} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     * @param onEvent the callback to call with the terminal event tuple
+     * @return the new Maybe instance
+     */
+    public final Maybe<T> doOnEvent(BiConsumer<? super T, ? super Throwable> onEvent) {
```

Comment:
But by that argument we should remove `doOnEach` from `Observable` and `Flowable` as well in favor of a method with this signature. At least then there'd be parity for peeking across all stream types.

I would much prefer a `doOnEach` in all the stream types and force those who really need zero allocation peeking to just do a `lift` or use the `doOnEach` overload which accepts individual `Consumer`s.